test/import/FSHImporter.Profile.test.ts

```diff
@@ -2145,7 +2177,88 @@ describe('FSHImporter', () => {
+        * insert CardRuleSet(path with spaces, 1, *)
+        `;
+        importer.import([new RawFSH(input, 'Insert.fsh')]);
+
+        expect(stats.numError).toBe(1);
+        expect(loggerSpy.getLastMessage('error')).toMatch(
+          /Error parsing insert rule with parameterized RuleSet CardRuleSet/s
+        );
+        expect(loggerSpy.getLastMessage('error')).toMatch(/File: Insert\.fsh.*Line: 4/s);
+      });
+
+      it('should log one error when nested insert rules with parameters result in multiple parser errors in the generated RuleSets', () => {
+        const input = `
+        Profile: MyObservation
+        Parent: Observation
+        * note 0..1
+        * insert FirstRiskyRuleSet("Observation.id")
+        `;
+        importer.import([new RawFSH(input, 'Insert.fsh')]);
+        expect(stats.numError).toBe(1);
+        expect(loggerSpy.getLastMessage('error')).toMatch(
+          /Errors parsing insert rule with parameterized RuleSet FirstRiskyRuleSet/s
+        );
+        expect(loggerSpy.getLastMessage('error')).toMatch(/File: Insert\.fsh.*Line: 5/s);
+      });
+
+      it('should not log an error when an insert rule with parameters results in rules that are syntactically correct but semantically invalid', () => {
+        const input = `
+        Profile: MyObservation
+        Parent: Observation
+        * insert CardRuleSet(nonExistentPath, 7, 4)
+        `;
+        const allDocs = importer.import([new RawFSH(input, 'Insert.fsh')]);
+        expect(allDocs).toHaveLength(1);
+        const doc = allDocs[0];
+
+        expect(doc.appliedRuleSets.size).toBe(1);
+        const appliedRuleSet = doc.appliedRuleSets.get(
+          JSON.stringify(['CardRuleSet', 'nonExistentPath', '7', '4'])
+        );
+        expect(appliedRuleSet).toBeDefined();
+        // This rule is nonsense, of course, but figuring that out is the job of the exporter, not the importer.
+        assertCardRule(appliedRuleSet.rules[0], 'nonExistentPath', 7, '4');
+        assertCardRule(appliedRuleSet.rules[1], 'note', 7, '4');
+        expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
+      });
+
+      it('should log one warning when an insert rule with parameters results in warnings', () => {
+        const input = `
+        Profile: MyObservation
+        Parent: Observation
+        * insert WarningRuleSet(Device)
+        `;
+        importer.import([new RawFSH(input, 'Insert.fsh')]);
+
+        expect(stats.numWarn).toBe(1);
+        expect(loggerSpy.getLastMessage('warn')).toMatch(
+          /Warnings parsing insert rule with parameterized RuleSet WarningRuleSet/s
+        );
+        expect(loggerSpy.getLastMessage('warn')).toMatch(/File: Insert\.fsh.*Line: 4/s);
+      });
+
+      it('should log one error when an insert rule with parameters results in non-parser errors', () => {
+        const input = `
+        Profile: MyObservation
+        Parent: Observation
+        * insert CardRuleSet(nonExistentPath, , )
```

Comment:
I tried some manual testing with this test case, and one issue I've noticed is that when an `InsertRule` references a `RuleSet` that is not defined, we end up adding `undefined` to the list of rules for that profile, so we end up with an error like this 
![image](https://user-images.githubusercontent.com/52747882/102643376-2da42d00-412d-11eb-9f32-3807f979985c.png)

Reply:
This FSH alone:
```
Profile: MyObservation
Parent: Observation
* insert CardRuleSet(nonExistentPath, , )
```
should recreate this.

Reply:
Ah, I understand, since this is now a scenario where a rule visitor might not return an instance of a rule. Do you think it would make more sense to return the `InsertRule` anyway and instead emit the error when applying the rule? This lines up with the timing for errors for non-parameterized rule sets, but it does mean that we would need to be keeping parameterized rule sets in their original forms on the documents in our tank so that we can correctly discern between "wrong number of parameters" and "undefined ruleset" errors for parameterized rule sets. The alternative to this is to modify some of the code in the importer so that `undefined` doesn't end up in a list of rules.

Reply:
I think if it's easier to implement, it's fine to just not add the rule to the list of rules at all. It's a bit of a break from what we have done with other rules, but at the same time this is a unique case.